src/core/server/plugins/plugins_system.ts

```diff
@@ -75,7 +78,10 @@ export class PluginsSystem {
       this.startedPlugins.push(pluginName);
     }
 
-    return exposedValues;
+    return {
+      startedPlugins: pluginStartContracts,
+      uiPluginDependencies: this.getTopolgociallySortedUiPluginDependencies(),
+    };
```

Comment:
@azasypkin Curious if you have any opposition to this change of the PluginSystem/PluginService. I wanted to expose just enough information to inject into the client-side to do what it needs. My only hesitation is this seems a bit specific and a bit more coupled than I would like, but that might be fine in this case. Thoughts?

Reply:
As I get further into the second phase of the UI Plugins (loading and starting them), I'm noticing that I will need more than just the pluginId and the list of dependencies. I think I will change this to export a `Map<string, UIPlugin>` where `UIPlugin` is a simplified interface derived from `Plugin`:

```ts
interface UIPlugin {
  id: string;
  dependencies: ReadonlyArray<string>; // all required deps + optional deps that are present
  configPath: ConfigPath;
}
```

Reply:
I'm not opposed to the idea general, haven't looked too much into implementation though. Previously I was thinking about returning something like `manifests: Map<PluginName, PluginManifest(Like)>` as a part of `PluginService`'s start contract where `manifests` would be topologically sorted so that `LegacyService` can decide what exactly to pass to the `KbnServer` as a part of constructor `core` argument (e.g. extract only UI plugins etc.).

Reply:
Updated to expose the entire manifest.